src/dns/aardvark.rs

```diff
@@ -0,0 +1,256 @@
+use crate::network::types;
+use nix::sys::signal::{self, Signal};
+use nix::unistd::Pid;
+use std::collections::HashMap;
+use std::fs;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::prelude::*;
+use std::io::BufReader;
+use std::io::Result;
+use std::net::IpAddr;
+use std::net::Ipv4Addr;
+use std::path::Path;
+
+#[derive(Debug, Clone)]
+pub struct AardvarkEntry {
+    pub network_name: String,
+    pub network_gateway_v4: String,
+    pub network_gateway_v6: String,
+    pub container_id: String,
+    pub container_ip_v4: String,
+    pub container_ip_v6: String,
+    pub container_name: Vec<String>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Aardvark {
+    // aardvark's config directory
+    pub config: String,
+}
+
+impl Aardvark {
+    pub fn new(config: String) -> Self {
+        Aardvark { config }
+    }
+
+    // On success retuns aardvark server's pid or returns -1;
+    fn get_aardvark_pid(self) -> i32 {
+        let pid: i32;
+        let path = Path::new(&self.config).join("aardvark.pid".to_string());
+        match fs::read_to_string(&path) {
+            Ok(content) => {
+                pid = match content.parse::<i32>() {
+                    Ok(val) => val,
+                    Err(_) => {
+                        return -1;
+                    }
+                }
+            }
+            Err(_) => {
+                return -1;
+            }
+        }
+        pid
+    }
+
+    pub fn notify(self) -> Result<()> {
+        let aardvark_pid = self.get_aardvark_pid();
+        if aardvark_pid != -1 {
+            signal::kill(Pid::from_raw(aardvark_pid), Signal::SIGHUP)?;
+        } else {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Invalid pid to notify",
+            ));
+        }
+
+        Ok(())
+    }
+    pub fn commit_entries(self, entries: Vec<AardvarkEntry>) -> Result<()> {
```

Comment:
We need to be able to handle "updates" here as well.  For example, network connect, where an alias might be added.

Reply:
This is not needed. A network connect/disconnect will always change the full network membership never a single alias.

Reply:
@baude This should be handled by `Setup(` already.